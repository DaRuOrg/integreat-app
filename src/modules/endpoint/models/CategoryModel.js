// @flow

type categoryModel = {
  id: number,
  url: string,
  title: string,
  parentId: number,
  parentUrl: string,
  content: string,
  thumbnail: string,
  order: number,
  availableLanguages: any
}

class CategoryModel {
   _id: number
  _url: string
  _title: string
  _parentId: number
  _parentUrl: string
  _content: string
  _thumbnail: string
  _order: number
  _availableLanguages: any

  constructor (categoryModel: categoryModel) {
    this._id = categoryModel.id
    this._url = categoryModel.url
    this._title = categoryModel.title
    this._content = categoryModel.content
    this._parentId = categoryModel.parentId
    this._parentUrl = categoryModel.parentUrl
    this._thumbnail = categoryModel.thumbnail
    this._order = categoryModel.order
    this._availableLanguages = categoryModel.availableLanguages
  }

  get thumbnail (): string {
    return this._thumbnail
  }

  get id (): number {
    return this._id
  }

  get url (): string {
    return this._url
  }

  get title (): string {
    return this._title
  }

  get content (): string {
    return this._content
  }

  get parentId (): number {
    return this._parentId
  }

  get parentUrl (): string {
    return this._parentUrl
  }

  setParentUrl (parentUrl: string) {
    this._parentUrl = parentUrl
  }

  get order (): number {
    return this._order
  }

  get availableLanguages (): any {
    return this._availableLanguages
  }
}

export default CategoryModel
