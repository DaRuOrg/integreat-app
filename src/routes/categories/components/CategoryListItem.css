@import url('../../../modules/common/constants/colors.css');

.row {
  display: flex;
  align-items: center;
  margin-bottom: 10px;
}

.category-thumbnail {
  width: 50px;
  height: 50px;
  flex-shrink: 0;
  padding: 5px;
  object-fit: contain;
}

.category-caption {
  height: 100%;
  min-width: 1px; /* needed to enabled line breaks for to long words, exact value doesn't matter, @Max: DO NOT CHANGE */
  flex-grow: 1;
  margin-left: 15px;
  padding: 15px 0;
  border-bottom: 2px solid var(--theme-color);
  word-wrap: break-word;
}
