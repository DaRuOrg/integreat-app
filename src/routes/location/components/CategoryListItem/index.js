import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'redux-little-router'

import style from './style.css'

import IconPlaceholder from './assets/IconPlaceholder.svg'

class CategoryListItem extends React.Component {
  static propTypes = {
    page: PropTypes.instanceOf(PageModel).isRequired,
    url: PropTypes.string.isRequired
  }

  render () {
    return (
      <Link href={this.props.url}>
        <div className={style.row}>
          <img className={style.categoryThumbnail} src={this.props.page.thumbnail || IconPlaceholder}/>
          <div className={style.categoryCaption}>{this.props.page.title}</div>
        </div>
      </Link>
    )
  }
}

export default CategoryListItem
