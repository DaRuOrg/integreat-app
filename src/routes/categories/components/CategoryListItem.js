import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'redux-little-router'

import CategoryModel from 'modules/endpoint/models/CategoryModel'

import style from './CategoryListItem.css'
import IconPlaceholder from '../assets/IconPlaceholder.svg'

/**
 * Displays a single CategoryListItem
 */
class CategoryListItem extends React.Component {
  static propTypes = {
    category: PropTypes.instanceOf(CategoryModel).isRequired
  }

  render () {
    return (
      <Link href={this.props.category.url}>
        <div className={style.row}>
          <img className={style.categoryThumbnail} src={this.props.category.thumbnail || IconPlaceholder} />
          <div className={style.categoryCaption}>{this.props.category.title}</div>
        </div>
      </Link>
    )
  }
}

export default CategoryListItem
