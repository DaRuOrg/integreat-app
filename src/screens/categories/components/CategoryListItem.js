// @flow

import * as React from 'react'

import CategoryModel from 'modules/endpoint/models/CategoryModel'
import iconPlaceholder from '../assets/IconPlaceholder.svg'
import styled from 'styled-components'

const Row = styled.View`
  margin: 12px 0;
`

const SubCategory = styled.View`
`

const CategoryThumbnail = styled.Image`
  width: 40px;
  height: 40px;
  flex-shrink: 0;
  padding: 8px;
`

const CategoryCaption = styled.Text`
  height: 100%;
  flex-grow: 1;
  padding: 15px 5px;
  border-bottom-width: 2px;
  border-bottom-color: ${props => props.theme.colors.themeColor};
`

const SubCategoryCaption = styled(CategoryCaption)`
  padding: 8px 0;
  border-bottom-width: 1px;
  border-bottom-color: ${props => props.theme.colors.themeColor};
`

const StyledLink = styled.View`
  align-items: center;
  margin: 0 auto;
`

type PropsType = {
  category: CategoryModel,
  subCategories: Array<CategoryModel>,
  /** A search query to highlight in the category title */
  query?: string
}

/**
 * Displays a single CategoryListItem
 */
class CategoryListItem extends React.Component<PropsType> {
  renderSubCategories (): Array<React.Node> {
    const {subCategories} = this.props
    return subCategories.map(subCategory =>
      <SubCategory key={subCategory.id}>
        <StyledLink to={subCategory.path}>
          <SubCategoryCaption search={''}>
            {subCategory.title}
          </SubCategoryCaption>
        </StyledLink>
      </SubCategory>
    )
  }

  renderTitle (): React.Node {
    const {query} = this.props
    return <CategoryCaption search={query || ''}>
      {this.props.category.title}
    </CategoryCaption>
  }

  render () {
    const {category} = this.props
    return (
      <Row>
        <StyledLink to={category.path}>
          <CategoryThumbnail source={category.thumbnail ? {uri: category.thumbnail} : iconPlaceholder} />
          {this.renderTitle()}
        </StyledLink>
        {this.renderSubCategories()}
      </Row>
    )
  }
}

export default CategoryListItem
