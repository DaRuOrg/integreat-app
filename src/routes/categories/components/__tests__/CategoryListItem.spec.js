import React from 'react'
import { shallow } from 'enzyme'

import CategoryModel from 'modules/endpoint/models/CategoryModel'
import CategoryListItem from '../CategoryListItem'

const category = new CategoryModel({
  id: 3649,
  url: '/augsburg/de/willkommen',
  title: 'Willkommen',
  content: 'this is a test content',
  parentId: 0,
  parentUrl: '/augsburg/de',
  order: 11,
  availableLanguages: {
    en: 4804, ar: 4819, fa: 4827
  },
  thumbnail: 'https://cms.integreat-apâ€¦03/Beratung-150x150.png'
})

describe('CategoryListItem', () => {
  test('should render and match snapshot', () => {
    const wrapper = shallow(
      <CategoryListItem category={category} children={[]} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
