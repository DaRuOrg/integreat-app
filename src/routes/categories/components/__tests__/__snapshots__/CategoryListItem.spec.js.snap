// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryListItem should render and match snapshot 1`] = `
<Connect(Link)
  href="/augsburg/de/willkommen"
>
  <div
    className="row"
  >
    <img
      className="categoryThumbnail"
      src="https://cms.integreat-apâ€¦03/Beratung-150x150.png"
    />
    <div
      className="categoryCaption"
    >
      Willkommen
    </div>
  </div>
</Connect(Link)>
`;
